uwblueprint/planet-read
# Create TranslationPage for translating story content

## Notion ticket link

[Load all story contents into cells/table](https://www.notion.so/uwblueprintexecs/Load-all-story-contents-into-cells-table-9d7c0ffb11954bb4a05c85b18920d717)



## Implementation description
(Commits should make it easier to review code and pieces and is an ok description of implementation:)
* Add components for page: image icons, `Cell` (story content), `EditableCell` (story translation content)
* Create `TranslationPage` to house these components. About how to store the story content / translation content -> array was the easier choice as it's ordered, maps easily, and passing content id / line Index around requests meant you could set quickly using indices (without looping to find). Back half of the `StoryLine` object is optional as translated content set in a different loop from story content.
* There's a couple bugs around story translation content queries/mutations, this fixes one of them (that it wasn't saving the changes into db when called)


## Steps to test
1. From the stories page, go to a translations route -> pressing `view translation / edit translation / translate book` should lead to a `/translation/storyId/storyTranslationId` route ... some of these pages are broken
2. Edit text boxes to make the translation. You can only do this slowly - the changes don't register properly when you do it quickly. However, you can change it not one letter at a time by copy/pasting longer messages. Note that the icons flicker (they are appear randomly, without any logic yet).
3. You can check that the changes are being saved properly: the easiest way is to refresh the page and ensure that the changes made still appear (and are thus saved somewhere). You can also open up the db and check that changes are being registered properly.
![translation load content](https://user-images.githubusercontent.com/40974372/125179893-4fe0c380-e1c1-11eb-85b4-193d9d7dc92b.gif)

`<TextArea>` doesn't grow automatically, but reportedly `<Input>` doesn't do multi-line 



## What should reviewers focus on?
* I stayed fairly light on the CSS. Do we have finalized designs or is this mostly placeholder until the stylized components to finish up?
* More CSS: I used flex for rows instead of grid... is that a bad decision from a best practices / efficiency standpoint?
* Is there a preferred name for the translation page route?
* (I think we talked about this, but just noting -> this re-renders the entire page for every change and makes a new query every single time. That seems less than ideal)
* How do the dimensions look on your browser? 
* Should create a ticket to investigate and double check which queries/mutations are broken? I'm seeing some null errors with `StoryTranslationsAvailableForReview` and `StoryTranslationsByUser` 
* Are there error states to cover, and do we want a loading card?

## Checklist
